### PR TITLE
Add tensorboard for gradient monitoring

### PR DESCRIPTION
**Pull request by Larwive**
Looks like we have some gradients to fix.

To run the tensorboard server:
`tensorboard --logdir runs`

Either create the `runs` directory before, or run `src/main.py` to let the directory be created automatically.
For now, tensorboard will be enabled by default, and will log gradients on every step. This should be the case until we know gradients work.

### CODE REVIEW

File: src/train.py [hunk 1]
```diff
@@ -23,18 +55,29 @@ def step(model, state, optimizer, device, is_image_based, action_space, reward):
...
     total_loss.backward()
     optimizer.step()
 
     return output_dict["action"], total_loss.item()
 
 
-def train(model: WorldModel, interface: GymEnvInterface, max_iter=10000, device='cpu'):
+def train(model: WorldModel, interface: GymEnvInterface, max_iter=10000, device='cpu', use_tensorboard: bool = True):
     print(model.parameters)
     optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
```
[the remark this answers is not in the document]
Comment (Larwive) [reply src/train.py:75]: 0.01 is an arbitrary value I set for now. I agree it should become a parameter but since nothing is working for the moment, I wanted to keep the code easy to understand.

**Review by Bugz-gg — APPROVED** [review]

**Review by Bugz-gg — APPROVED** [review]
